diff --git a/meson.build b/meson.build
index 899a017d67f9..b067c6a1a5ce 100644
--- a/meson.build
+++ b/meson.build
@@ -99,6 +99,17 @@ zlib_dep       = dependency('zlib')
 jpeg_dep       = dependency('libjpeg')
 png_dep        = dependency('libpng', version : '>1.2')
 openjpeg_dep   = dependency('libopenjp2', version : '>=2.1.0')
+if host_machine.system() == 'windows'
+  openjpeg_dep = declare_dependency(
+    dependencies : [
+      cc.find_library('openjp2'),
+      openjpeg_dep.partial_dependency(
+        compile_args : true,
+        includes : true,
+      )
+    ]
+  )
+endif
 tiff_dep       = dependency('libtiff-4')
 glib_dep       = dependency('glib-2.0', version : '>=2.56')
 gio_dep        = dependency('gio-2.0')
